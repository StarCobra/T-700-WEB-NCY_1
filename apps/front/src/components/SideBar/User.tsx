import React from "react";
import profilePic from "../../assets/sideBar/profil.png";
import { Box } from "@mui/material";
import { Link } from "react-router-dom";

export default function User(props: any) {
  const { user } = props;

  return (
    <Link to="/profile" className="userLocation">
      <Box className="userProfilePicture">
        <img src={profilePic} alt="" />
      </Box>

      <Box className="usernameContainer">
        <Box className="userName">{`${user?.first_name} ${user?.last_name}`}</Box>
        <Box className="userMail">{user?.email}</Box>
      </Box>
    </Link>
  );
}
