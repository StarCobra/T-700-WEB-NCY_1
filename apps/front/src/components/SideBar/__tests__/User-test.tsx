import React from "react";
import { act, create } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import User from "../User";

describe("Components > SideBar > User", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(
          <MemoryRouter>
            <User {...test.props} />
          </MemoryRouter>,
        );
      });
      expect(root).toMatchSnapshot();
    });
  });
});
