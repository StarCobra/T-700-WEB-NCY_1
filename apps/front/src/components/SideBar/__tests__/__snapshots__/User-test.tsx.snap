// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > SideBar > User should render the component generic 1`] = `
<a
  className="userLocation"
  href="/profile"
  onClick={[Function]}
>
  <div
    className="userProfilePicture MuiBox-root css-0"
  >
    <img
      alt=""
      src="default"
    />
  </div>
  <div
    className="usernameContainer MuiBox-root css-0"
  >
    <div
      className="userName MuiBox-root css-0"
    />
    <div
      className="userMail MuiBox-root css-0"
    />
    ;
  </div>
</a>
`;
